### PR TITLE
use service files from caddy repo

### PR DESCRIPTION
**Pull request by computerlyrik**
- copy over service files directly from caddy vendor
- some cleanup in main task

### CODE REVIEW

**Comment by lbischof** [comment]
Thanks for the pull request. I think it's a great idea to use the official files, but I think we still need to fix a few things. 

- First the tests are failing because it can't start the service. The tests run two times and after the first run they shouldn't change the service status, because it should already be running.
- What happens to the email? Currently the email is added to the exec in the service file so that lets encrypt works automatically. Is there another way to do this? maybe in the configfile?
- Is the working directory correct? Currently we use /home/caddy but I doubt the official service files use the same paths. We probably need to create them and set the correct permissions.

Maybe we should copy the files and edit them accordingly (email etc). That way we can also test new changes and updates. What do you think?

**Comment by computerlyrik** [comment]
@lbischof  Have a look :)

**Comment by computerlyrik** [comment]
Commit https://github.com/antoiner77/caddy-ansible/pull/18/commits/59d340900f147cb60090eaedd2c025a97de4dd5a
contains lineinfile fix for
https://github.com/mholt/caddy/issues/822

File: tasks/main.yml [hunk 1]
```diff
@@ -55,5 +58,22 @@
...
+    regexp: '^ExecStart\='
+    line: 'ExecStart=/usr/bin/caddy -log stdout -agree=true -email {{ caddy_email }} -conf=/etc/caddy/Caddyfile -root=/var/www'
+  when: not caddy_init_system or caddy_init_system == "systemd"
+  notify: Reload systemd service file
+
+- name: Systemd service | fix port permission error
+  lineinfile: 
+    state: absent
+    dest: /etc/systemd/system/caddy.service
+    regexp: '^NoNewPrivileges\=true'
```
Comment (lbischof) [inline tasks/main.yml]: What is this? Why are you removing it?

**Comment by lbischof** [comment]
I would prefer copying the service files, instead of downloading and changing lines.

File: tasks/main.yml [hunk 2]
```diff
@@ -2,33 +2,36 @@
...
   when: caddy_update
   register: caddy_features_cache
 
 - name: Download Caddy
-  get_url: url=https://caddyserver.com/download/build?os=linux&arch=amd64&features={{caddy_features}} dest=/home/caddy/caddy.tar.gz force=yes
+  get_url: url=https://caddyserver.com/download/build?os=linux&arch=amd64&features={{caddy_features}} dest={{ caddy_home }}/caddy.tar.gz force=yes
   when: caddy_releases_cache.changed or caddy_features_cache.changed
   register: caddy_binary_cache
 
-- name: Download Caddy
```
Comment (lbischof) [inline tasks/main.yml]: Why are you removing this? Please create a new pull request.

File: tasks/main.yml [hunk 3]
```diff
@@ -2,33 +2,36 @@
...
-  get_url: url=https://caddyserver.com/download/build?os=linux&arch=amd64&features={{caddy_features}} dest=/home/caddy/caddy.tar.gz
-
 - name: Extract Caddy
-  unarchive: src=/home/caddy/caddy.tar.gz dest=/usr/bin/ copy=no
+  unarchive: src={{ caddy_home }}/caddy.tar.gz dest=/usr/bin/ creates=/usr/bin/caddy copy=no
   when: caddy_binary_cache.changed
 
-- name: Extract Caddy
```
Comment (lbischof) [inline tasks/main.yml]: Why are you removing this? Please create a new pull request.

**Comment by computerlyrik** [comment]
Hi.
About the removing line: there should be a fix not using setcap anymore. see https://github.com/mholt/caddy/issues/822 - i did not test it yet

About the Exec-Line: It is following the howto on https://github.com/mholt/caddy/blob/master/dist/init/linux-systemd/README.md "Configure the service"

File: tasks/main.yml (hunk 2 above)
Comment (computerlyrik) [inline tasks/main.yml]: Why is this code double in your mainline?

**Comment by computerlyrik** [comment]
I know, using the vendor service files means a larger refactoring.

And as soon, as we have a final agreed version, i am willing to do different branches and set different pull requests.

File: tasks/main.yml (hunk 2 above)
Comment (lbischof) [inline tasks/main.yml]: it's not double, the first download block only get's executed when the versions or features change and this only happens when auto_update is set to true. The second block downloads caddy if it doesn't exist.

Comment (computerlyrik) [inline tasks/main.yml]: Okay. I will remove this in a clean pull request.
How about the changes in the service files?

If you really want to copy them over, i can add a condition to vars with default from template. And let the user decide if they want the github&replace-way.

**Comment by lbischof** [comment]
Please copy the service files and edit them accordingly. I would like to keep this module simple and testable. I'm not exactly sure what advantage there is from downloading them each time, except losing control. This pull request should be a simple change: Add the new service files. Further changes can be discussed in the future.

**Comment by wmark** [comment]
Yes, the email address can be given in the configfile (```Caddyfile```).

File: tasks/main.yml [hunk 4]
```diff
@@ -39,27 +42,38 @@
...
+    regexp: '^ExecStart\='
+    line: 'ExecStart=/usr/bin/caddy -log stdout -agree=true -email {{ caddy_email }} -conf=/etc/caddy/Caddyfile -root=/var/www'
+  when: not caddy_init_system or caddy_init_system == "systemd"
+  notify: Reload systemd service file
+
+- name: Systemd service | fix port permission error
+  lineinfile: 
+    state: absent
+    dest: /etc/systemd/system/caddy.service
+    regexp: '^NoNewPrivileges\=true'
```
Comment (wmark) [inline tasks/main.yml]: I doubt this is needed. I guess computerlyrik experienced a bug with xenolf/lego, has worked around said bug, and this happened to be an artifact of his prior testing. At least that's my speculation based on having tried to reproduce the original report on Gentoo and Debian Jessie.

Comment (lbischof) [inline tasks/main.yml]: What distributions use such an old version of systemd? Should we support them?

Comment (computerlyrik) [inline tasks/main.yml]: Currently Debian 8 Jessie - see
https://github.com/mholt/caddy/issues/822#issuecomment-219252967

**Comment by lbischof** [comment]
Closing this pull request in favor of #19

**Pull request closed without merging**